client/state/data-layer/wpcom-http/utils.js

```diff
@@ -46,18 +51,18 @@ export const getHeaders = action => get( action, 'meta.dataLayer.headers', undef
  */
 export const getProgress = action => get( action, 'meta.dataLayer.progress', undefined );
 
-class SchemaError extends Error {
+export class SchemaError extends Error {
 	constructor( errors ) {
 		super( 'Failed to validate with JSON schema' );
 		this.schemaErrors = errors;
 	}
 }
 
-class TransformerError extends Error {
-	constructor( error, transformer, data ) {
+export class TransformerError extends Error {
+	constructor( error, data, transformer ) {
 		super( error.message );
-		this.transformer = transformer;
 		this.inputData = data;
+		this.transformer = transformer;
 	}
 }
```

Comment:
fun fact! I'm second-guessing my inclusion of a transformer error class. we can `throw` _anything_ from inside of there, which I think can be a valid way to signal different kinds of failures.

if it's a schema validation error we can check `instanceof SchemaError` or `isinstance NetworkError` but for the general `fromApi()` interface I can imagine common `throw`s of things like `false` or actual data, such as a some error object - things that exception classes may not be necessary for

(this is a comment not related to this PR)